app/src/NativeBase/Components/NavigationHeader.tsx

```diff
@@ -27,7 +27,7 @@ const NavigationHeader: React.FC<StackHeaderProps> = ({
     <>
       <StatusBar barStyle={statusBarStyle} />
 
-      <Box safeAreaTop>
```

Comment:
This caused some additional whitespace in iOS. Safe Area in Navigators/Application is enough to render the headers as expected.